### DISCUSSION TITLE
"zoxide: command not found" error on bash shell instance initialization, despite zoxide being accessible in $PATH

### DISCUSSION THREAD

**Discussion by dashelm** [ajeetdsouza/zoxide#705] [Q&A] [answered]
Hello!
Just installed zoxide (Ubuntu Server 20.04.6 LTS), and I am having some trouble getting bash to play nice with it.

Namely, the eval statement provided in the documentation to be placed at the end of ~/.bashrc is causing "zoxide: command not found" to be printed whenever a new shell instance is created. The process is clearly not being started.

Entering "which zoxide" returns:
`/home/dashel/.local/bin/zoxide`

and echo $PATH returns:
`/home/dashel/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin:/home/dashel/.dotnet/tools`

I'm able to use the "zoxide" command, proving that zoxide definitely exists _somewhere_, but the "z" command does not work.

zoxide's install location is included in $PATH, so why might bash be unable to find zoxide? And why might the "z" command be broken?
Thanks!

**Comment by AbdifitahW** [comment]
I'm facing this same issue and only once sourcing my .bashrc file does the z command function again.

**Comment by ajeetdsouza** [comment, marked as answer]
Somehow, it seems like `~/.local/bin` is being added to your `$PATH` _after_ you call `eval`.

Try adding `export PATH="$HOME/.local/bin:$PATH"` just before you call`eval` in your `.bashrc`.

**Reply by dashelm** [reply]
This worked!
Thank you so much for the help.

**Reply by abhishekchoure** [reply]
This worked for me as well. (Ubuntu WSL)

**Comment by jkottler** [comment]
In a similar situation, using Ubuntu 22.04 on WSL2, I found that the answer was to put the eval line at the end of `~/.profile` instead of `~/.bashrc`.
